src/languages/go.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  createPatternMatchers,
+  argumentMatcher,
+  leadingMatcher,
+  conditionMatcher,
+  trailingMatcher,
+  childMatcher,
+  cascadingMatcher,
+  patternMatcher,
+} from "../util/nodeMatchers";
+import {
+  NodeMatcherAlternative,
+  ScopeType,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { getNodeRange } from "../util/nodeSelectors";
+import { SyntaxNode } from "web-tree-sitter";
+
+// Generated by the following command:
+// `curl https://raw.githubusercontent.com/tree-sitter/tree-sitter-go/master/src/node-types.json | jq '[.[] | select(.type == "_statement" or .type == "_simple_statement") | .subtypes[].type]'`
+const STATEMENT_TYPES = [
+  "_expression",
+  "assignment_statement",
+  "dec_statement",
+  "inc_statement",
+  "send_statement",
+  "short_var_declaration",
+  "_simple_statement",
+  "block",
```

Comment:
I think we want to drop this one.  @AndreasArvidsson just fixed a bug in Typescript where if you are in an empty if statement, saying "take state" will take only the curly brackets when you really want the whole if statement.  It's worth recording a test case like the following:

```go
if true {
    |
}
```

where `|` is your cursor position, then saying "clear state".  It should result in an empty file

```suggestion
```

Reply:
Yeah the current behavior is wrong, and looks like removing `block` fixes it, thanks!